internal/services/containers/container_group_resource.go

```diff
@@ -857,14 +857,41 @@ func resourceContainerGroupUpdate(d *pluginsdk.ResourceData, meta interface{}) e
 		model.Identity = expandedIdentity
 	}
 
-	if d.HasChange("tags") {
-		model.Tags = tags.Expand(d.Get("tags").(map[string]interface{}))
+	// As API doesn't return the value of StorageAccountKey, so it has to get the value from tf config and set it to request payload. Otherwise, the Update API call would fail
+	addedEmptyDirs := map[string]bool{}
+	_, initContainerVolumes, err := expandContainerGroupInitContainers(d, addedEmptyDirs)
+	if err != nil {
+		return err
+	}
+	_, _, containerVolumes, err := expandContainerGroupContainers(d, addedEmptyDirs)
+	if err != nil {
+		return err
+	}
+	var containerGroupVolumes []containerinstance.Volume
+	if initContainerVolumes != nil {
+		containerGroupVolumes = initContainerVolumes
 	}
+	if containerGroupVolumes != nil {
+		containerGroupVolumes = append(containerGroupVolumes, containerVolumes...)
+	}
+	model.Properties.Volumes = pointer.To(containerGroupVolumes)
 
+	// As Update API doesn't support to update identity, so it has to use CreateOrUpdate API to update identity
 	if err := client.ContainerGroupsCreateOrUpdateThenPoll(ctx, *id, model); err != nil {
 		return fmt.Errorf("updating %s: %+v", *id, err)
 	}
```

Comment:
If only the tags are updated then we would be making an unnecessary update call. I think this whole block should be moved into 
```
if d.HasChange("identity") {
...
}
```

Reply:
Updated